app/vmagent/remotewrite/remotewrite.go

```diff
@@ -485,6 +484,15 @@ func tryPush(at *auth.Token, wr *prompbmarshal.WriteRequest, forceDropSamplesOnF
 			rowsCountAfterRelabel := getRowsCount(tssBlock)
 			rowsDroppedByGlobalRelabel.Add(rowsCountBeforeRelabel - rowsCountAfterRelabel)
 		}
+		idxDiff := 0
+		for i, ts := range tssBlock {
+			if storagelimits.ExceedingLabels(ts.Labels) {
+				idxDiff++
+			} else if idxDiff > 0 {
+				tssBlock[i-idxDiff] = tssBlock[i]
+			}
+		}
+		tssBlock = tssBlock[:len(tssBlock)-idxDiff]
```

Comment:
It's better to check if len(tssBlock)==0 afterwards, and skip following steps if it is.

Reply:
Looks like more idiomatic way to filter in-place: 
```
# slice tricks https://go.dev/wiki/SliceTricks#filter-in-place
n := 0
for _, x := range a {
    if keep(x) {
        a[n] = x
        n++
    }
}
a = a[:n]
```

 At VictoriaMetrics code base more often tmpSlice pattern is used:
```
tmpA := a[:0]
for _, x := range a {
    if keep(x) {
        tmpA = append(tmpA,x)
    }
}
a = tmpA
```